src/components/MovieDetails/MovieDetails.js

```diff
@@ -1,38 +1,64 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import './MovieDetails.css'
+import './MovieDetails.css';
+import { Link } from 'react-router-dom';
+import { fetchMovieInfo } from '../../apiCalls';
 
-const MovieDetails = ({movieInfo, resetClickedMovie}) => {
-    console.log('what is here', movieInfo.tagline)
-    const formattedRating = movieInfo.average_rating.toFixed(0);
-
-    const formatOverviewLength = () => {
-        if (movieInfo.overview.length > 182) {
-           const shortOverview = movieInfo.overview.slice(0, 182) + '...';
-           return shortOverview;
-        } else {
-            return movieInfo.overview;
+class MovieDetails extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            movieInfo: {}
         }
     }
-    // const formattedDate = 
-    return (
-        <section className="movieDetailsContainer" style={{ backgroundImage: `url(${movieInfo.backdrop_path})`}}>
-                <button className='returnHome' onClick={() => {resetClickedMovie()}}>X</button>
-            <div className='movieDescription'>
-                <img className='movie-poster' src={movieInfo.poster_path} id={movieInfo.id} alt={'Cover art image for ' + movieInfo.title}></img>
-                <div className='movieDetailsCard'>
-                    <h2>{movieInfo.title}</h2>
-                    <h4>{movieInfo.tagline}</h4>
-                    <p>{"Genre: " + movieInfo.genres}</p>
-                    <p>{"Rating: " + formattedRating + "/10"}</p>
-                    <p>{movieInfo.release_date}</p>
-                    <p className='overview'>{formatOverviewLength()}</p>
+   
+    componentDidMount() {
+        console.log("heyyyyyyyy", this.props.id);
+        // this.props.updateClickedMovie(this.props.movieInfo.id);
+        fetchMovieInfo(this.props.id)
+            .then(movie => this.setState({ movieInfo: movie.movie }))
+            .catch(error => this.setState({ error: 'Having trouble finding this movie right now...please try again.'} ));
+
+    }
+
+    // formatOverviewLength = () => {
+    //     if (this.props.movieInfo.overview.length > 182) {
+    //         const shortOverview = this.props.movieInfo.overview.slice(0, 182) + '...';
+    //         return shortOverview;
+    //     } else {
+    //         return this.props.movieInfo.overview;
+    //     }
+    // }
+    render() {
+        // const formattedDate = 
+        console.log("DO YOU RENDER")
```

Comment:
I think we're finally on the right track here.  I think if we add in error handling for the page loading we should be able to accomplish what we're trying to do (without having to pass in all of the movie's info to the render.  ComponentDidMount() should take care of that for us after the first render()